metastore/src/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java

```diff
@@ -1518,6 +1518,15 @@ void addDynamicPartitions(long txnId, String dbName, String tableName, List<Stri
   void insertTable(Table table, boolean overwrite) throws MetaException;
 
   /**
+   * Performs the pre insert operation before the job starts
+   * @param table table name
+   * @param overwrite true if the insert is overwrite
+   *
+   * @throws MetaException
+   */
+  void preInsertTable(Table table, boolean overwrite) throws MetaException;
```

Comment:
@ashutoshc would this also to the function InsertTable added in the previous PR ?